rct229/rulesets/ashrae9012019/section5/section5rule53.py

```diff
@@ -0,0 +1,238 @@
+                index_rmr="baseline",
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def is_applicable(self, context, data=None):
+            # Building level check if any doors in the building
+            building_p = context.baseline
+            subsurfaces = find_all(
+                "$.building_segments[*].zones[*].surfaces[*].subsurfaces[*]", building_p
+            )
+            return any(
+                [
+                    subsurface_b.get("classification") == DOOR
+                    and subsurface_b.get("glazed_area", ZERO.AREA)
+                    >= subsurface_b.get("opaque_area", ZERO.AREA)
+                    for subsurface_b in subsurfaces
+                ]
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.baseline
+            climate_zone = data["climate_zone"]
+            return {
+                "surface_conditioning_category_dict_b": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                )
+            }
+
+        def list_filter(self, context_item, data=None):
+            # context_item shall be the list of the list_path element
+            surface_b = context_item.baseline
+            # exclude unregulated surfaces
+            return (
+                data["surface_conditioning_category_dict_b"][surface_b["id"]]
+                != SCC.UNREGULATED
+            )
+
+        class SurfaceRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule53.BuildingRule.SurfaceRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule53.BuildingRule.SurfaceRule.SubSurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="$.subsurfaces[*]",
+                    required_fields={
+                        "$.subsurfaces[*]": [
+                            "classification",
+                            "glazed_area",
+                            "opaque_area",
+                        ]
+                    },
+                )
+
+            def create_data(self, context, data):
+                surface_b = context.baseline
+                return {
+                    "surface_conditioning_category_b": data[
+                        "surface_conditioning_category_dict_b"
+                    ][surface_b["id"]]
+                }
+
+            def list_filter(self, context_item, data):
+                subsurface_b = context_item.baseline
+                # exclude non door surfaces and doors whose opaque area is greater than glazed area
+                return (
+                    subsurface_b["classification"] == DOOR
+                    and subsurface_b["glazed_area"] >= subsurface_b["opaque_area"]
+                )
+
+            class SubSurfaceRule(RuleDefinitionBase):
```

Comment:
The implementation doesn't seem to match the RDS. Can the RDS be updated?

Reply:
The RDS is now updated and approved